packages/@react-stately/calendar/src/useCalendarState.ts

```diff
@@ -169,6 +174,35 @@ export function useCalendarState<T extends DateValue = DateValue>(props: Calenda
   }, [calendarDateValue, isDateUnavailable, minValue, maxValue]);
   let validationState = props.validationState || (isUnavailable ? 'invalid' : null);
 
+  let pageDuration = useMemo(() => {
+    if (pageBehavior === 'visible') {
+      return visibleDuration;
+    }
+
+    if (visibleDuration.days) {
+      return {
+        days: 1
+      };
+    }
+
+    if (visibleDuration.weeks) {
+      return {
+        weeks: 1
+      };
+    }
+
+    if (visibleDuration.months) {
+      return {
+        months: 1
+      };
+    }
+    
+    return {
+      years: 1
+    };
```

Comment:
There is a `unitDuration` helper function which should take care of this.

Reply:
Thanks! I've updated the code.